In [13]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split, cross_validate, GridSearchCV, learning_curve

## Load Data

In [14]:
# Load updated df
df = pd.read_csv("../Data/df_updated.csv")

In [15]:
df.shape

(5265, 33)

In [16]:
df.head().T

0               1               2  \
Unnamed: 0                             0               1               2   
Skin symptoms                          1               1               1   
Season born                          1.0             2.0             2.0   
Heating system_house                 1.0             2.0             1.0   
House _now                           NaN             2.0             3.0   
House_before                         NaN             3.0             1.0   
Clean_house                          2.0             2.0             2.0   
Puppy_dewormed                       1.0             1.0             1.0   
Puppy_vaccinated                     1.0             1.0             1.0   
Dam_dewormed_prebirth                NaN             1.0             1.0   
Dam_vaccinated_prebirth                0               0               0   
Gender                               1.0             2.0             2.0   
Over 50% white                       2.0             2.0             2.0   
Other animals                          1               1               2   
Other dogs                             1               2               2   
Born in owner family                   1               2               1   
Does the dog have a yard             2.0             2.0             2.0   
Bcs_under 2 mo                       NaN             1.0             3.0   
Outside_under 2 mo                   5.0             1.0             1.0   
Outside a day_under 5mo              NaN             3.0             3.0   
Skin symptoms_dam                    1.0             1.0             1.0   
Age                                  7.0             2.0            13.0   
Vet diagnosis                        0.0             0.0             0.0   
Breed_En                   Border Collie   Border Collie   Border Collie   
Classification            Brightest Dogs  Brightest Dogs  Brightest Dogs   
Obey                                95.0            95.0            95.0   
Reps_lower                           1.0             1.0             1.0   
Reps_upper                           4.0             4.0             4.0   
Height_low_inches                     19              19              19   
Height_high_inches                    21              21              21   
Weight_low_lbs                        40              40              40   
Weight_high_lbs                       40              40              40   
Category                         herding         herding         herding   

                                       3               4  
Unnamed: 0                             3               4  
Skin symptoms                          1               1  
Season born                          1.0             4.0  
Heating system_house                 2.0             1.0  
House _now                           4.0             3.0  
House_before                         NaN             1.0  
Clean_house                          2.0             2.0  
Puppy_dewormed                       1.0             1.0  
Puppy_vaccinated                     1.0             1.0  
Dam_dewormed_prebirth                1.0             1.0  
Dam_vaccinated_prebirth                0               0  
Gender                               2.0             2.0  
Over 50% white                       2.0             2.0  
Other animals                          1               2  
Other dogs                             2               2  
Born in owner family                   1               1  
Does the dog have a yard             2.0             2.0  
Bcs_under 2 mo                       1.0             1.0  
Outside_under 2 mo                   1.0             1.0  
Outside a day_under 5mo              2.0             NaN  
Skin symptoms_dam                    1.0             NaN  
Age                                  1.0            10.0  
Vet diagnosis                        0.0             0.0  
Breed_En                   Border Coll

In [17]:
df.drop('Unnamed: 0', axis=1, inplace=True)

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5265 entries, 0 to 5264
Data columns (total 32 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Skin symptoms             5265 non-null   int64  
 1   Season born               5140 non-null   float64
 2   Heating system_house      4762 non-null   float64
 3   House _now                5238 non-null   float64
 4   House_before              3930 non-null   float64
 5   Clean_house               5239 non-null   float64
 6   Puppy_dewormed            4998 non-null   float64
 7   Puppy_vaccinated          5080 non-null   float64
 8   Dam_dewormed_prebirth     3524 non-null   float64
 9   Dam_vaccinated_prebirth   5265 non-null   int64  
 10  Gender                    5124 non-null   float64
 11  Over 50% white            4991 non-null   float64
 12  Other animals             5265 non-null   int64  
 13  Other dogs                5265 non-null   int64  
 14  Born in 

In [11]:
main_label = 'Skin symptoms'
extra_label = 'Vet diagnosis'

In [12]:
y = df[main_label].values.reshape(-1,)
X = df.drop([main_label, extra_label], axis=1) # drop extra labels
cat_cols = X.select_dtypes(include=['object']).columns
cat_cols_idx = [list(X.columns).index(c) for c in cat_cols]
X_train,X_test,y_train,y_test = train_test_split(X, y, test_size=0.3, random_state=40)
X_train.shape, X_test.shape, y_train.shape, y_test.shape

Index(['Breed_En', 'Classification', 'Height_low_inches', 'Height_high_inches',
       'Weight_low_lbs', 'Weight_high_lbs', 'Category'],
      dtype='object')


((3685, 30), (1580, 30), (3685,), (1580,))

## Summary 
In this step we split the data 70%/30% train and test which provided a training set of 3685 rows / 30 columns and a test set of 1580 rows / 30 columns. 